### DISCUSSION TITLE
where to change width of the component row ?

### DISCUSSION THREAD

**Discussion by mikevanoostende** [templaza/astroid-framework#249] [Q&A]
Hi,

I've got another issue with a template I want to install using the Astroid framework,
as I'm in urgent need to upgrade to PHP 8..

As you can see in the pictures below I have by default all products lined up under each other, I expect them to be 4 in 1 line.
with my older "joom" template everything was fine.
I checked all settings in the new template, but I cannot find where to remove this "width = 100%", is this a VM parameter ? or still something related to my template (I use tp_adventure from joomlaplates)
as this is already compiled I do not know how to find the source.
Any help would be appreciated .

I'm using VirtueMart 4.0.4 10688

thx, Mike.

![Schermafbeelding 2022-11-20 om 17 52 58](https://user-images.githubusercontent.com/3439472/203020187-ca454f81-df93-4447-8868-98d2ab2fe1a2.jpg)
![Schermafbeelding 2022-11-20 om 17 53 08](https://user-images.githubusercontent.com/3439472/203020190-2714c431-73cb-416d-a07c-5a748cf972bd.jpg)

**Comment by Chacapamac** [comment]
Looking at this, you are probably using **VirtueMart**. 
Any change of template can bring stuff like that as you bringing new library to the mix.
 
— Here we can see that VirtueMart use it’s **own column version** (vm-col) but sadly didn’t use their own specific class for the (.row) with the same name that the (.row) in Bootstrap that come with Astroid causing probably the problem.

**I see two solutions:**
1. Maybe you have a setting in Virtuemart to use Bootstrap in their shop in place of their own code ? 
2. You need to overrule the width: 100% only for that specific situation in your custom css.
**— My first try with this will be:**
div.browse-view div.row { width: inherit;} or div.browse-view div.row { width: inherit !important;}
or
div.browse-view .row { width: inherit;} or div.browse-view .row { width: inherit !important;}

If your website is online, I can take a look...

Good Day!

**Reply by mikevanoostende** [reply]
Hi, 
thx for your answer, I would appreciate it if you have a look,
currently I only set up 1 page with this template to check before I'll switch it as being the default, this is the link : 
https://passion4wood.be/nl/download-files/download-pers-passion4wood-verlichting/hang-en-plafond-verlichting-in-hout-2.html

thx, Mike.

**Comment by Chacapamac** [comment]
body.view-category div.product {width:25%;}

— I add the body class .view-category to be more specific to that view of VirtueMart
— It seem to work for smaller screen to and go to 3 or 2 or 1 for phone view
— 25% give you four products on a row

I’m almost sure that you can control this in Virtuemart settings, you should ask in their forum of their documentation.
It make no sens to override if you can control it by a setting...

**Reply by Chacapamac** [reply]
Can you tell me if it work?

**Reply by mikevanoostende** [reply]
Hi, it didn't work last time, but as I was in a hurry i installed another template, but I'll change it back soon to test your solution again. is there a new version available ? 
thx Mike.

**Comment by Chacapamac** [comment]
I test the CSS directly in the link that you give me with my browser inspector , with whatever template where install at that point and it work correctly.

**Last version of Astroid is v2.6.3**
https://github.com/templaza/astroid-framework/releases/tag/v2.6.3

![Screen Shot 2022-12-20 at 5 19 15 PM](https://user-images.githubusercontent.com/1907276/208778437-79449f24-bc89-4794-a6d8-5a179691032d.jpg)

**Comment by Chacapamac** [comment]
@mikevanoostende 
I see that you are not using Astroid anymore on you example, can we close that topic?
